src/tea_html.ml

```diff
@@ -224,8 +224,8 @@ let class' name = prop "className" name
 
 let classList classes =
   classes
-  |> List.filter (fun (_fst, snd) -> snd)
-  |> List.map (fun (fst, _snd) -> fst)
+  |. Belt.List.keep (fun (_fst, snd) -> snd)
```

Comment:
What on earth is `keep`?  I've never heard of a `filter` function being called `keep` in any language I've used in 30 years...  o.O

Any chance of using a proper descriptive name?

Also, `|.` is... odd looking, what does it gain here?

Reply:
I agree with `keep` -> `filter`. Belt functions have the data structure as the first argument which doesn't work well with a sequence of `|>`. `|.` was added as an alternative. `a |. b c` = `b a c`.

Reply:
Except I can't seem to find where `|.` is defined.  The data structure should be at the end, not the start, else currying because quite a bit more difficult (why on earth put it at the start in a currying language?!?).

Reply:
> Except I can't seem to find where `|.` is defined.

I think it's part of the syntax? See https://github.com/BuckleScript/bucklescript/pull/2620.

> The data structure should be at the end, not the start, else currying because quite a bit more difficult (why on earth put it at the start in a currying language?!?).

Yes, even with `|.` I think this might not be a great idea. Now some APIs will use `|>` and some will use `|.` resulting in (IMO) unnecessary complexity as can be seen in this diff itself.

Reply:
> I think it's part of the syntax? See BuckleScript/bucklescript#2620.

Uh, then how would that work with the native compiler?  Or is BuckleScript becoming a PP as well as a PPX layer on the native compiler?  ^.^;

> Yes, even with |. I think this might not be a great idea. Now some APIs will use |> and some will use |. resulting in (IMO) unnecessary complexity as can be seen in this diff itself.

Yeah I was noticing the odd mixture, it was rubbing me the wrong way but I figured it was just because I didn't understand what the purpose of `|.` even is, but if it is just to put an argument at the front instead of the end then that just yells to me that the API that you are using it with is *badly* misdesigned...

Reply:
(I might be wrong so let's wait for @bobzhang to reply!)

Reply:
Also, since it prepends it, then what happens if something like this is done:

```ocaml
let a b c d = b + c + d

let aa = a 1

let aaa = 2 |. aa 3
```

Will `aaa` end up calling `a 2 1 3` as I'd expect if it stuffs in front or would it end up calling `a 1 2 3`, which I'd might also expect considering that's more textual though not really correct for the definition?

Things like `|>` that just adds an argument to the current curry set is trivial...

Reply:
Tested both of these:

```ocaml
let aaaa =
  let a b c d = b + c * d in
  2 |. a 1 3

let aaaa' =
  let a b c d = b + c * d in
  2 |. (a 1) 3
```

And they return different results...  That is *very* wtf'y...  I'm not liking this `|.` operator at all...  :-(

Reply:
I'm not sure why you wouldn't expect different results here. Just from looking at it, I think would expect `aaaa to be: a 2 1 3` and `aaaa' to be a 1 2 3`

To be clear, I'm not sure if I like the `|.` yet, but I'm willing to try it. Elixir has only currying like this. Furthermore, there were some arguments in the proposal (https://github.com/facebook/reason/issues/1452)

Reply:
Elixir doesn't have partial application though.  For languages without partial application then front-first piping makes sense (and I actually prefer it this way as it reads more naturally as well), but for languages *with* partial application (like OCaml) then piping should always only ever pipe into the current pending (end) position.  So someone even proposing an Elixir-like piping operator into OCaml does not know the language and does not understand the ramifications (like the `2 |. a 1 3` and `2 |. (a 1) 3` differences is an obvious design fault, obvious 'why' it happens, but it is surprising nonetheless as it just should not happen in a language with partial application).

Reply:
Actually, if the purpose of front-filling was just for JS devs side stuff, then why doesn't bucklescript 'flip' the arguments in all functions and calls when it compiles to javascript?  That way the final 'data' argument in OCaml is the first argument in Javascript (thus also making named and optional arguments later in the Javascript argument list as well, which is indeed inverted from the OCaml standard style because OCaml has partial application).